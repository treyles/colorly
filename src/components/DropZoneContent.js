import React from 'react';
import PropTypes from 'prop-types';

export default class DropZoneContent extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    this.props.handleImageDrop();
  }

  render() {
    const { imageAlert } = this.props;

    return (
      <div className="drop-zone-content">
        <label htmlFor="file-browse" className={imageAlert && 'alert'}>
          {imageAlert ? (
            <h1>{imageAlert}</h1>
          ) : (
            <div>
              <h1>Drag and Drop Image Here</h1>
              <h3>JPG, GIF, PNG or SVG no larger than 3 MB</h3>
            </div>
          )}
        </label>
        <input
          id="file-browse"
          type="file"
          accept="image/*"
          onChange={this.handleChange}
        />
      </div>
    );
  }
}

DropZoneContent.defaultProps = {
  imageAlert: null
};

DropZoneContent.propTypes = {
  imageAlert: PropTypes.string,
  handleImageDrop: PropTypes.func.isRequired
};
