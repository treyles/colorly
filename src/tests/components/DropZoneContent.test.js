import React from 'react';
import { shallow, mount } from 'enzyme';
import DropZoneContent from '../../components/DropZoneContent';

describe('DropZoneContent', () => {
  const props = {
    imageAlert: false,
    handleImageDrop: jest.fn()
  };

  it('renders correctly', () => {
    const component = shallow(<DropZoneContent {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders alert message based on imageAlert', () => {
    const component = mount(<DropZoneContent {...props} />);

    component.setProps({
      imageAlert: 'test alert'
    });

    expect(component.find('label h1').text()).toEqual('test alert');
  });
});
