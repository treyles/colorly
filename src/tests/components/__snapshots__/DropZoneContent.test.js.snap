// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropZoneContent renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropZoneContent
    handleImageDrop={[MockFunction]}
    imageAlert={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="false"
          htmlFor="file-browse"
        >
          <div>
            <h1>
              Browse or Drop Image Here
            </h1>
            <h3>
              JPG, GIF, PNG or SVG no larger than 3 MB
            </h3>
          </div>
        </label>,
        <input
          accept="image/*"
          id="file-browse"
          onChange={[Function]}
          type="file"
        />,
      ],
      "className": "drop-zone-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <h1>
              Browse or Drop Image Here
            </h1>
            <h3>
              JPG, GIF, PNG or SVG no larger than 3 MB
            </h3>
          </div>,
          "className": "false",
          "htmlFor": "file-browse",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Browse or Drop Image Here
              </h1>,
              <h3>
                JPG, GIF, PNG or SVG no larger than 3 MB
              </h3>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Browse or Drop Image Here",
              },
              "ref": null,
              "rendered": "Browse or Drop Image Here",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "JPG, GIF, PNG or SVG no larger than 3 MB",
              },
              "ref": null,
              "rendered": "JPG, GIF, PNG or SVG no larger than 3 MB",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "accept": "image/*",
          "id": "file-browse",
          "onChange": [Function],
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="false"
            htmlFor="file-browse"
          >
            <div>
              <h1>
                Browse or Drop Image Here
              </h1>
              <h3>
                JPG, GIF, PNG or SVG no larger than 3 MB
              </h3>
            </div>
          </label>,
          <input
            accept="image/*"
            id="file-browse"
            onChange={[Function]}
            type="file"
          />,
        ],
        "className": "drop-zone-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <h1>
                Browse or Drop Image Here
              </h1>
              <h3>
                JPG, GIF, PNG or SVG no larger than 3 MB
              </h3>
            </div>,
            "className": "false",
            "htmlFor": "file-browse",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Browse or Drop Image Here
                </h1>,
                <h3>
                  JPG, GIF, PNG or SVG no larger than 3 MB
                </h3>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Browse or Drop Image Here",
                },
                "ref": null,
                "rendered": "Browse or Drop Image Here",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "JPG, GIF, PNG or SVG no larger than 3 MB",
                },
                "ref": null,
                "rendered": "JPG, GIF, PNG or SVG no larger than 3 MB",
                "type": "h3",
              },
            ],
            "type": "div",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "accept": "image/*",
            "id": "file-browse",
            "onChange": [Function],
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
